### PR TITLE
DM-25807: Suppress known deprecated Filter calls

### PR DESCRIPTION
**Pull request by parejkoj**
I did this as part of DM-25807 because the extra warnings from obs_cfht in jointcal were making it harder to debug those tests.

### CODE REVIEW

File: python/lsst/obs/cfht/megacamMapper.py [hunk 1]
```diff
@@ -109,10 +114,12 @@ def __init__(self, **kwargs):
...
                                        + MegacamMapper._nbit_filter)
-
-        if len(afwImageUtils.Filter.getNames()) >= 2**MegacamMapper._nbit_filter:
-            raise RuntimeError("You have more filters defined than fit into the %d bits allocated" %
-                               MegacamMapper._nbit_filter)
+        with warnings.catch_warnings():
+            # surpress Filter warnings; we already know this is deprecated
+            warnings.simplefilter('ignore', category=FutureWarning)
+            if len(afwImageUtils.Filter.getNames()) >= 2**MegacamMapper._nbit_filter:
+                raise RuntimeError("You have more filters defined than fit into the %d bits allocated" %
```
Comment (natelust) [inline python/lsst/obs/cfht/megacamMapper.py:121]: Please don't use % based formatting

**Review by natelust — APPROVED** [review]

File: python/lsst/obs/cfht/megacamMapper.py (hunk 1 above)
Comment (parejkoj) [reply python/lsst/obs/cfht/megacamMapper.py:121]: I don't want to modify the existing code, it just got tabbed-in one level due to the added context manager.

Comment (natelust) [reply python/lsst/obs/cfht/megacamMapper.py:121]: That's fair

**Review by natelust — APPROVED** [review]

**Comment by parejkoj** [comment]
It looks like @timj already did this on #81 , so there's nothing to merge here.

**Comment by timj** [comment]
Sorry. I didn't know there was another ticket for it and they were annoying me.

**Pull request closed without merging**